A post about the dynamite package

helske opened the pull request:
* [x] I have read the Content Guidelines.
* [x] I have read the Technical Guidelines.
* [x] I used or followed the R Markdown or Markdown template.
* [x] I have followed the Style Guide.
* [x] I created or updated my author metadata with correct folder name.
* [x] I have added relevant tags after browsing existing tags (including "community" tag).
* [x] I have added the "tech notes" tag if this is a technote.
* [x] I ran `roblog::ro_lint_md()` on index.md (optional).
* [x] I ran `roblog::ro_check_urls()` on index.md (optional).
* [x] I ran a spell-check on index.md.
* [x] I have added the tags - Software Peer Review, my-packagename.
* [x] I have added the package-version YAML tag.
* [x] I have added acknowledgement of the reviewers' work (with links to reviewers).
* [x] I have added a link to the software peer review thread.

Here's an initial draft post (I guess this is more in line with tech notes than blog post?) about the [`dynamite`](https://docs.ropensci.org/dynamite/index.html) package. I try to get the package hex logo ready after holiday and hopefully at that time the package is also at CRAN (already submitted) so I can add a link to that as well.

steffilazerte commented:
Hi @helske, I'm just getting caught up on the backlog while I was on holidays. 
Thanks for creating a post!
When you are ready for review, just give me a ping and I'll take a look and we can figure out a date for publishing :grin:

helske commented:
Hi @steffilazerte I made some tweaks to the post and I think its now ready for a review. There is one issue with the post though, at least locally on my computer the math does not render properly, but perhaps (hopefully) it is just my setup?

maelle commented:
:wave: @steffilazerte @helske! I'm investigating how we can render the math (#473). Congrats @helske on writing the first math formulas of the whole blog. :smile_cat: 

In the meantime, do not worry about it. No matter what the final solution looks like, I'll be the one tweaking the post as necessary. 

(@steffilazerte I opened https://github.com/ropensci-org/blog-guidance/issues/212)

steffilazerte commented on content/blog/2022-12-15-dynamite-r-package/index.Rmd:
````diff
@@ -0,0 +1,334 @@
+p1 + p2 + p3 + plot_layout(ncol = 1)
+```
+
+As expected, the (wrong) conclusion would be that our original intervention did not affect $y$.
````
```suggestion
As expected, because we did not allow multiple response variables to vary, the (wrong) conclusion would be that our original intervention did not affect $y$.
```

steffilazerte reviewed (commented):
Hi @helske! Thank you for submitting a post for the blog (and congrats on finishing peer review!)

Your post is really nicely written with clear instructions to the reader how to follow you. dynamite is so sleek looking and seems really powerful and it was lovely to see how you guided the reader through the examples.

The main way I think this post could be tweaked is to add some more basic explanations up front to users who may be less familiar with these kinds of stats. For example, I'm generally familiar with statistics used in biology, but had no clue what things like "interventional distributions" or "synthetic control" were. You definitely don't have teach the reader these things, but if you could add in an occasional sentence (much like you already have in many places) helping readers who may be less familiar, I think it'll go a long way to making this article accessible to a broader audience.

I have added 28 suggestions/comments, but most are tiny fiddly edits or places where I think a bit of explanation could be useful. The only things that really should be fixed are the Alt text (see my suggestions, they should be more descriptive), the stylistic items, and the occasional typo. Everything else is just a suggestion, so don't hesitate to take it as is, modify it completely, or reject it outright :grin:

What do you think of a publication date of Tuesday Jan 31st or Thursday Feb 2? If you can get the edits back to me by Friday, we can definitely do Tuesday, or if you'd like a couple more days, and can get the edits to me by Monday, we can publish on Thursday. We can also push back publication to a week later, there's no rush 😄. 
When we have a date, don't forget to modify both the yaml data *and* the folder name.

[X] post follows Content Guidelines
[X] post follows Style Guide - (**No formatting for package names**)
[X] title is in Title Case
[ ] publication date is ok - **How about 2023-01-31 or 2023-02-02?**
[ ] alternative text of images is informative - **See my suggestions**
[X] author metadata is provided with correct folder name
[X] html not included in pull request of Rmd post
[X] I ran `roblog::ro_lint_md()` on [index.md](http://index.md/)
[X] I ran `roblog::ro_check_urls()` on [index.md](http://index.md/)
[X] I ran a spell-check on [index.md](http://index.md/)
[X] YAML subject tags are ok ("tech notes" for tech notes; "community" for non-staff non-editor)
[X] YAML field 'preface' is present if necessary
[X] YAML package-version included
[X] YAML subject tags - software peer review, packagename
[X] acknowledges and links to reviewers
[X] links to peer review thread

helske commented:
Thanks @steffilazerte, I will look into these probably tomorrow, so publishing in Jan 31st should be ok.

maelle commented:
@steffilazerte @helske please ping me (here + Slack if possible) once the post is ready or ready-ish, and I'll do the katex stuff at that point so that you might check formulas look good.

helske commented:
@steffilazerte (and ping to @maelle) Thanks again for your very helpful and nice comments, I have now modified the text/code according to your suggestions. Your comment on expanding the synthetic control was especially helpful, as writing this made me realize that I had slightly misunderstood the original idea of the method and my text could have been bit confusing (it's hopefully less confusing now). Fortunately it did not affect the main message of the post.

maelle commented:
On it now, please do not touch the PR in the meantime :smile_cat: :construction:

maelle commented:
@helske does it look good to you? Thanks for being the beta tester of math support on our blog :smile:

helske commented:
Thanks @maelle, looks good, I added `echo=FALSE` to remove katex code from being visible in the output and fixed the list in the beginning where I had apparently dropped an empty line earlier today.

helske commented:
Hmm, I'm not sure what happened, looks like the figures are not visible anymore.

maelle commented:
ah I thought it was maybe due to my Pandoc version but no, some weird interaction between the things. A quick fix is to remove the line breaks between "{{" and "<figure" and ">" and "}}" (that's what I did, I might explore more later).

maelle commented:
I have explored a bit but haven't found a solution yet (apart from fixing manually), I'm sorry but will come back to it at some point.

steffilazerte commented:
Great thanks @helske! I'll take a formal look tomorrow

maelle commented:
@helske I found a fix, now knitting should no longer break the figures. Thank you for your patience!

maelle commented:
https://deploy-preview-459--ropensci.netlify.app/

steffilazerte reviewed (commented):
Hi @helske 

The added details are perfect! I have a much better understanding of how this kind of modelling works (and it's fascinating). I think this is now much more accessible, thank you!

I've noted a couple more extremely minor suggestions. Mostly small grammar tweaks that I missed the first time around and a couple of formatting bits. 

If you can have a look at those suggestions and make any you would like by Tuesday Jan 31st at 15:00 UTC (my morning when I'll go to publish), that'd be great!

Also, nice to see all the math rendered correctly (thanks @maelle!).

Finally, I'll tweet/toot an announcement of the post. Below I have a draft, let me know if you have any comments/suggestions.

Thanks again!

**Twitter:**

```
[technote] Dynamite for Causal Inference from Panel Data using Dynamic Multivariate Panel Models

@jouni_helske explains the fascinating world of synthetic controls and interventions with clear examples exploring the power of dynamite.

https://ropensci.org/blog/2023/01/31/dynamite-r-package/

#rstats

(1/2)


Our first blog post on a package to go through statistical review! 🎉🎉🎉

🙏 Thank yous to 

rOpenSci reviewers: @nj_clark and @LucyStats
dynamite package coauthor: Santtu Tikka
Project: @PREDLIFEproject

(2/2)
```

------------------------
**Mastodon**

```
[technote] Dynamite for Causal Inference from Panel Data using Dynamic Multivariate Panel Models

@jouni_helske@fediscience.org explains the fascinating world of synthetic controls and interventions with clear examples exploring the power of dynamite.

https://ropensci.org/blog/2023/01/31/dynamite-r-package/

Our first blog post on a package to go through statistical review! 🎉🎉🎉

🙏 Thank yous to 

rOpenSci reviewers: @nj_clark@twitter.com and @lucystats@mastodon.social
dynamite package coauthor: Santtu Tikka
Project: @PREDLIFEproject@twitter.com
```

helske commented on content/blog/2023-01-31-dynamite-r-package/index.Rmd:
```diff
@@ -0,0 +1,368 @@
+for `r katex::katex_html("t=2, \\ldots, 100", displayMode = FALSE)`, where the initial values `r katex::katex_html("x_1", displayMode = FALSE)` and `r katex::katex_html("y_1", displayMode = FALSE)` are drawn from the standard normal distribution (in subsequent modelling we condition on these first observations, i.e. they are treated as fixed data).
+
+Variable `r katex::katex_html("z_t", displayMode = FALSE)` is our intervention variable, which I fixed to zero for the first 80 time points, and to one for the the last 20 time points, i.e. the intervention starts at time `r katex::katex_html("t=81", displayMode = FALSE)`.
+Following the terminology in the DMPM paper, this is a recurring intervention, in contrast to an atomic intervention where `r katex::katex_html("z_{81}=1", displayMode = FALSE)` and zero otherwise.
```
Thanks, I mean the socarxiv preprint, adding link is a good idea but apparently not that easy to accomplish: https://github.com/jgm/pandoc/issues/1603. Not sure what would have been the best solution, I just added ^1 without hyperlink.

helske commented:
Thanks again for both of you @steffilazerte and @maelle for all the tweaks and suggestions! The tweets/toots look good as well.

steffilazerte commented:
Great!
Btw I pushed a small fix for that link that was giving your trouble, how does [this](https://deploy-preview-459--ropensci.netlify.app/blog/2023/01/31/dynamite-r-package/#data-generation) look to you?

helske commented:
Oh yeah that works!

steffilazerte commented:
Posted 🎉 !
Thanks again 🙏🏼 !
